internal/operator-controller/applier/helm_test.go

```diff
@@ -32,6 +34,28 @@ type mockPreflight struct {
 	upgradeErr error
 }
 
+type noOpPreAuthorizer struct{}
+
+func (p *noOpPreAuthorizer) PreAuthorize(
+	ctx context.Context,
+	ext *ocv1.ClusterExtension,
+	manifestReader io.Reader,
+) ([]authorization.ScopedPolicyRules, error) {
+	// No-op: always return an empty map and no error
+	return nil, nil
+}
+
+type errorPreAuthorizer struct{}
+
+func (p *errorPreAuthorizer) PreAuthorize(
+	ctx context.Context,
+	ext *ocv1.ClusterExtension,
+	manifestReader io.Reader,
+) ([]authorization.ScopedPolicyRules, error) {
+	// Always returns no missing rules and an error
+	return nil, errors.New("problem running preauthorization")
```

Comment:
For follow-up: we should do two things here:
1. Consolidate `noOpPreAuthorizer` and `errorPreAuthorizer` into a single struct (maybe `mockPreAuthorizer`?) that has fields for `missingRules` and `err`. Then populate them according to the desired test scenario
2. We should have tests at this level for:
    - missingRules: nil, err: non-nil -> error result includes `err` (we have this now)
    - missingRules: non-nil, err: nil -> error result includes missing rules
    - missingRules: non-nil, err: non-nil -> error result includes missing rules and `err`
    - missingRules: nil, err: nil -> no error (we have this now)

Reply:
I'll add an issue so we remember to come back and do it?

Reply:
Follow-up issue https://github.com/operator-framework/operator-controller/issues/1923